assets/js/components/user-input/UserInputPreview.js

```diff
@@ -58,7 +57,11 @@ export default function UserInputPreview( { back, goTo } ) {
 		setIsNavigating( true );
 		const response = await saveUserInputSettings();
 		if ( ! response.error ) {
-			global.location.assign( addQueryArgs( dashboardURL, { notification: 'user_input_success' } ) );
+			const url = new URL( redirectURL || dashboardURL );
+			// Here we don't use `addQueryArgs` due to a bug with how it handles hashes
+			// See https://github.com/WordPress/gutenberg/issues/16655
+			url.searchParams.set( 'notification', 'user_input_success' );
```

Comment:
We don't need to add this query parameter for the alternative redirect, it should only apply to the default `dashboardURL`. That query parameter isn't used elsewhere.

That way we can simplify this to only `addQueryArgs` to the default URL like before (if no `redirectURL` is passed), and use `redirectURL` as is.